src/synthesizer/parametric/morphology.py

```diff
@@ -46,30 +51,64 @@ class Sersic2D(MorphologyBase):
     """
     A class holding the Sersic2D profile. This is a wrapper around the
     astropy.models.Sersic2D class.
-
-    Methods
-    -------
-    compute_density_grid
-        Calculates and returns the density grid defined by this Morphology
     """
 
     def __init__(
         self,
-        r_eff_kpc=None,
+        r_eff=None,
         r_eff_mas=None,
-        n=1,
-        ellip=0,
+        r_eff_kpc=None,
+        sersic_index=1,
+        ellipticity=0,
         theta=0.0,
         cosmo=None,
         redshift=None,
     ):
-        """ """
+        """
+        Initialise the morphology.
+        
+        Arguments
+            r_eff (unyt)
+                Effective radius. This is converted as required.
+            sersic_index (float)
+                Sersic index.
+            ellipticity (float)
+                Ellipticity.
+            theta (float)
+                Theta, the rotation angle.
+            cosmo (astro.cosmology)
+                astropy cosmology object.
+            redshift (float)
+                Redshift.
+
+        """
+        self.r_eff_mas = None
```

Comment:
You've left the arguments for these but then explicitly set them to None here. Probably worth just fully removing the specific arguments.

Reply:
I've removed the arguments but kept the attributes for now since it's useful to record both.

Reply:
Agreed, that's how `parametric.Stars` works with the single argument-multiple method.